zokrates_stdlib/stdlib/hashes/sha256/1024bitPadded.code

```diff
@@ -0,0 +1,15 @@
+import "./1536bit.code" as sha256
+// Take two field[256] arrays as input 
+// and returns their sha256 full round output as an array of 256 field elements.
+def main(field[256] a, field[256] b, field[256] c, field[256] d) -> (field[256]):
```

Comment:
Is it possible to have a generic version of the SHA256 function?

See: https://github.com/HarryR/ethsnarks/blob/master/src/utils.cpp#L80 - This is a `bits2blocks_padded` function, which converts an array of bits into a sequence of blocks with SHA256-compliant padding. The output of the function can be passed into a SHA256 chain.

Instead of having specific versions of the SHA2-256 function for different lengths, it would be better to have a single function which handles an arbitrary number of bits as input then adds padding as is necessary.

I think it would be a good exercise to see how this language can handle something like that.

Reply:
Absolutely, that's the goal. Not in this PR though!

One way that's feasible even in the short term is to allow for loop bounds to be any expression, and only make sure that these expressions are a compile time constant. Then you could write

`sha256(x, 42)` with 42 the number of rounds
but not `sha256(x, n)` where n is an input to the program